lib/puppet/parser/functions/noop.rb

```diff
@@ -1,16 +1,16 @@
-# Set noop to true as default for current and children scopes
-Puppet::Parser::Functions::newfunction(:noop, :doc => "Set noop default to true for all resources
+# Set noop to true as default for current and children scopes, or 'reset' an inherited default with noop(false)
+Puppet::Parser::Functions::newfunction(:noop, :doc => "Set noop default for all resources
   in local scope and children scopes. This can be overriden in
   child scopes, or explicitly on each resource.
   ") do |args|
-
+      @noop_value = true if (@noop_value = args[0]).nil? or (![true, false].include? @noop_value)
```

Comment:
Is the result of this code that when explicitly setting noop to "true" it will behave as though it has been explicitly passed "false"? Some basic IRB testing of this statement seems to suggest it will be.

I think I'd like to see some better definition of what acceptable arguments are and how they are treated. This should be better defined in the documentation and covered in tests.